command/base_flags.go

```diff
@@ -749,6 +750,134 @@ func (f *FlagSet) Var(value flag.Value, name, usage string) {
 	f.flagSet.Var(value, name, usage)
 }
 
+// -- TimeVar and timeValue
+type TimeVar struct {
+	Name       string
+	Aliases    []string
+	Usage      string
+	Default    time.Time
+	Hidden     bool
+	EnvVar     string
+	Target     *time.Time
+	Completion complete.Predictor
+	Formats    TimeFormat
+}
+
+// Identify the allowable formats, identified by the minimum
+// precision accepted.
+// TODO: move this somewhere where it can be re-used for the API.
+type TimeFormat int
+
+const (
+	TimeVar_EpochSecond TimeFormat = 1 << iota
+	TimeVar_RFC3339Nano
+	TimeVar_RFC3339Second
+	TimeVar_Day
+	TimeVar_Month
+)
+
+// Default value to use
+const TimeVar_TimeOrDay TimeFormat = TimeVar_EpochSecond | TimeVar_RFC3339Nano | TimeVar_RFC3339Second | TimeVar_Day
+
+// parseTimeAlternatives attempts several different allowable variants
+// of the time field.
+func parseTimeAlternatives(input string, allowedFormats TimeFormat) (time.Time, error) {
+	// The RFC3339 formats require the inclusion of a time zone.
+	if allowedFormats&TimeVar_RFC3339Nano != 0 {
+		t, err := time.Parse(time.RFC3339Nano, input)
+		if err == nil {
+			return t, err
+		}
+	}
+
+	if allowedFormats&TimeVar_RFC3339Second != 0 {
+		t, err := time.Parse(time.RFC3339, input)
+		if err == nil {
+			return t, err
+		}
+	}
+
+	if allowedFormats&TimeVar_Day != 0 {
+		t, err := time.Parse("2006-01-02", input)
+		if err == nil {
+			return t, err
+		}
+	}
+
+	if allowedFormats&TimeVar_Month != 0 {
+		t, err := time.Parse("2006-01", input)
+		if err == nil {
+			return t, err
+		}
+	}
+
+	if allowedFormats&TimeVar_EpochSecond != 0 {
+		i, err := strconv.ParseInt(input, 10, 64)
+		if err == nil {
+			// Four-digit numbers are ambiguous, don't
+			// parse them as a time in 1970.
+			if i > 10000 {
```

Comment:
Do we really care to accept anything older than say 2020?  What if someone decides to leave off date separators and gives 20200830 as input?

Reply:
That's a good use case to consider. We could eliminate everything below about 946000000.

Reply:
Changed bound to 946000000 (2000-ish)